pkg/compactor/bucket_compactor.go

```diff
@@ -670,6 +679,14 @@ func NewBucketCompactorMetrics(blocksMarkedForDeletion prometheus.Counter, reg p
 			Name: "cortex_compactor_group_compactions_total",
 			Help: "Total number of group compaction attempts that resulted in new block(s).",
 		}),
+		blockCompactionDelay: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+			Name:                            "cortex_compactor_block_compaction_delay_minutes",
+			Help:                            "Delay between a block being created and successfully compacting it in minutes.",
```

Comment:
nits:
* block being created => block being uploaded.
* I'd drop "in minutes" (or seconds) part of the sentence, that's already clear from metric name.